azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -0,0 +1,165 @@
+}
+# TODO: add 'arm64', 'aarch64' here once support is enabled for ARM64
+POLICY_SUPPORTED_ARCHITECTURE = ['x86_64']
+
+
+class PolicyEngine(object):
+    """
+    Implements policy engine API. Class will always be initialized, but if the Regorus import fails,
+    all methods will be no-ops.
+
+    If any errors are thrown in regorus.py, they will be caught and handled here. add_policy, add_data,
+    and set_input will be no-ops, eval_query will return an empty dict
+    """
+    def __init__(self, rule_file, policy_file):
+        """
+        rule_file: Path to a Rego file that specifies rules for policy behavior.
+
+        policy_file: Path to a JSON file that specifies parameters for policy behavior - for example,
+        whether allowlist or extension signing should be enforced.
+        The expected file format is:
+        {
+            "azureGuestAgentPolicy": {
+                "policyVersion": "0.1.0",
+                "signingRules": {
+                    "extensionSigned": <true, false>
+                },
+                "allowListOnly": <true, false>
+            },
+            "azureGuestExtensionsPolicy": {
+                "allowed_ext_1": {
+                    "signingRules": {
+                        "extensionSigned": <true, false>
+                    }
+                }
+        }
+        """
+        self._policy_engine_enabled = False
+        self._engine = None
+        if not self.is_policy_enforcement_enabled():
+            self.log_policy(msg="Policy enforcement is not enabled.")
+            return  # policy_engine_enabled is not set to True
+
+        if not self.is_policy_enforcement_supported():
+            # TODO: Update this error message when conf flag is removed post private preview.
+            msg = "Attempted to enable policy enforcement, but feature is not supported on this platform."
+            self.log_policy(msg=msg)
+            raise PolicyError(msg)
+
+        try:
+            self._engine = regorus.Engine()
+            self._engine.add_policy(rule_file)
+            self._engine.add_data(policy_file)
+            self._policy_engine_enabled = True
+        except Exception as ex:
+            msg = "Failed to initialize Regorus policy engine: '{0}'".format(ex)
+            self.log_policy(msg=msg)
+            raise PolicyError(msg)
+
+    @classmethod
+    def log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
+        """
+        Log information to console and telemetry.
+        """
+        if is_success:
+            logger.info("[Policy] " + msg)
+        else:
+            logger.error("[Policy] " + msg)
+        if send_event:
+            add_event(op=op, message=msg, is_success=is_success)
+
+    @staticmethod
+    def is_policy_enforcement_enabled():
+        """Check whether user has opted into policy enforcement feature"""
+        # TODO: Remove conf flag post private preview and add other checks here.
```

Comment:
We'll have a better idea after the discussion with PMs and v-team about user experience, but this change is relatively far-out since we're keeping the conf flag for now.

Reply:
Can you improve the comment a litte bit in that case? At this point it says that something needs to be done ("add other checks"), but not exactly what. Somebody trying to implement that TODO would be confused

Reply:
Updated